tests/testing.py

```diff
@@ -0,0 +1,312 @@
+
+
+class TransactionTests(Base):
+    amount = 5000000
+    denom = "atestfet"
+    msg_type = '/cosmos.bank.v1beta1.MsgSend'
+
+    def testNativeTransfer(self):
+
+        self.db_cursor.execute('TRUNCATE table native_transfers')
+        self.db.commit()
+
+        tx = self.ledger_client.send_tokens(self.delegator_wallet.address(), self.amount, self.denom, self.validator_wallet)
+        tx.wait_to_complete()
+        self.assertTrue(tx.response.is_successful())
+
+        time.sleep(5)
+
+        row = self.db_cursor.execute('SELECT * from native_transfers').fetchone()
+
+        self.assertIsNotNone(row)
+
+        self.assertEqual(row[1]['amount'][0]['amount'], str(self.amount))
+        self.assertEqual(row[1]['amount'][0]['denom'], self.denom)
+        self.assertEqual(row[1]['toAddress'], self.delegator_address)
+        self.assertEqual(row[1]['fromAddress'], self.validator_address)
+
+    def testRetrieveTransfer(self):
+        query = gql(
+            """
+            query getNativeTransfers {
+                nativeTransfers {
+                    nodes {
+                        transactionId
+                        message 
+                        }
+                    }
+                }
+            """
+        )
+
+        result = self.gql_client.execute(query)
+        self.assertEqual(result["nativeTransfers"]["nodes"][0]["message"]["amount"][0]["denom"], self.denom)
+        self.assertEqual(result["nativeTransfers"]["nodes"][0]["message"]["amount"][0]["amount"], str(self.amount))
+        self.assertEqual(result["nativeTransfers"]["nodes"][0]["message"]["toAddress"], self.delegator_address)
+        self.assertEqual(result["nativeTransfers"]["nodes"][0]["message"]["fromAddress"], self.validator_address)
+
+
+class LegacyBridgeSwapTests(Base):
+    contract = None
+    amount = "10000"
+    denom = "atestfet"
+
+    def setUp(self):
+        url = "https://github.com/fetchai/fetch-ethereum-bridge-v1/releases/download/v0.2.0/bridge.wasm"
+        if not os.path.exists("../.contract"):
+            os.mkdir("../.contract")
+        try:
+            temp = open("../.contract/bridge.wasm", "rb")
+            temp.close()
+        except:
+            contract_request = requests.get(url)
+            open("../.contract/bridge.wasm", "wb").write(contract_request.content)
+
+        self.contract = LedgerContract("../.contract/bridge.wasm", self.ledger_client)
+
+        self.contract.deploy(
+            {"cap": "250000000000000000000000000",
+             "reverse_aggregated_allowance": "3000000000000000000000000",
+             "reverse_aggregated_allowance_approver_cap": "3000000000000000000000000",
+             "lower_swap_limit": "1",
+             "upper_swap_limit": "1000000000000000000000000",
+             "swap_fee": "0",
+             "paused_since_block": 18446744073709551615,
+             "denom": "atestfet",
+             "next_swap_id": 0
+             },
+            self.validator_wallet
+        )
```

Comment:
I think we've realized that we don't need to deploy this if we're relying on oneshot bridge-related docker services to perform setup.